packages/lesswrong/components/users/DialogueSuggestionsPage.tsx

```diff
@@ -0,0 +1,782 @@
+  // const GET_USERS_DIALOGUE_CHECKS = gql`
+  //   query getUsersDialogueChecks {
+  //     getUsersDialogueChecks {
+  //       _id
+  //       __typename
+  //       userId
+  //       targetUserId
+  //       checked
+  //       match
+  //     }
+  //   }
+  // `;
+
+  // // useMulti fetching from dialogueChecks based on a view, which takes as a usedId as an input. 
+  // // permission stuff happens in the background
+  
+  // const { loading: loadingChecks, error: errorChecks, data: dataChecks } = useQuery(GET_USERS_DIALOGUE_CHECKS);  // for all the targetUsers thus obtained, check if there's a match 
+
+  let targetUserIds = [];
+  if (userDialogueChecks) {
+    targetUserIds = userDialogueChecks.map(check => check.targetUserId);
+  }
+
+  async function createDialogue(title: string, participants: string[]) {
+    const createResult = await createPost({
+      data: {
+        title,
+        draft: true,
+        collabEditorDialogue: true,
+        coauthorStatuses: participants.map(userId => ({userId, confirmed: true, requested: false})),
+        shareWithUsers: participants,
+        sharingSettings: {
+          anyoneWithLinkCan: "none",
+          explicitlySharedUsersCan: "edit",
+        },
+        contents: {
+          originalContents: {
+            type: "ckEditorMarkup",
+            data: ""
+          }
+        } as AnyBecauseHard
+      },
+    });
+    if (createResult?.data?.createPost?.data) {
+      const post = createResult?.data?.createPost?.data;
+      if (post) {
+        const postId = post._id;
+        const postEditUrl = `/editPost?postId=${postId}`;
+        history.push(postEditUrl);
+      }
+    }
+  }
+
+  if (!currentUser) return <p>You have to be logged in to view this page</p>
+  if (loading) return <p>Loading...</p>;
+  if (error || !userDialogueChecks || userDialogueChecks.length > 1000) return <p>Error </p>; // if the user has clicked that much stuff things might break...... 
+  if (userDialogueChecks?.length > 1000) {
+    console.log("Warning: userDialogueChecks.length > 1000, seems user has checked more than a thousand boxes? how is that even possible? let a dev know and we'll fix it...")
+  }
+
+  const handleOptInToRevealDialogueChecks = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    setOptIn(event.target.checked);
+    void updateCurrentUser({revealChecksToAdmins: event.target.checked})
+    captureEvent("optInToRevealDialogueChecks", {optIn: event.target.checked})
+    
+    const userDetailString = currentUser?.displayName + " / " + currentUser?.slug
+  
+    // ping the slack webhook to inform team of opt-in. YOLO:ing and putting this on the client. Seems fine. 
+    const webhookURL = "https://hooks.slack.com/triggers/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" // dev mode link for now, update before pushing to prod "https://hooks.slack.com/triggers/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";
```

Comment:
If you need a url conditional on environment, you can use an instance setting

Reply:
(If you want to ship without that, that's fine, just keep in mind you won't be able to distinguish dev & prod behavior then.)